### ISSUE TITLE
Request for a feature

### ISSUE THREAD

**Issue by skrishnan-planview** [vkolgi/mailprobe#3]
It's not an issue, rather a feature. Current setup does not differentiate from which environment the mail comes. If we can have some concept like 'Realms' or 'Environments' such that One Mailprobe instance can be shared with different mail sending services then it would be helpful. Right now, we have to install Mailprobe for each dev and different testing environments seperately.

Or if we can have a unique label mechanism also to differentiat the source instance, then it would be helpful

**Comment by vkolgi** [comment]
I agree we will need some sort of segregation mechanism here. If we have to segregate based on the mail sender instance, that information has to be sent as part of custom headers. One of the custom X headers has to be set. 
To make this feature generic, mailprobe will have a UI mapping facility where, if mail header has "<X-custom-header>" or "X-Originating-IP", then assign a label to it. Rest of the aggregation/filtering via APIs can consider the same label.

**Comment by skrishnan-planview** [comment]
ok, that sounds nice. Thanks for taking it up
